pkgs/development/tools/reshape/default.nix

```diff
@@ -0,0 +1,36 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, rustPlatform
+, fetchpatch
+, git
+, nixosTests
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "reshape";
+  version = "0.6.1";
+
+  src = fetchFromGitHub {
+    owner = "fabianlindfors";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-iX8qhDk0PP0AB3bJ6ck8AZ9SCErFH416ggAkgK8O900=";
+  };
+
+  cargoPatches = [
+    ./add-Cargo.lock.patch
+  ];
```

Comment:
You can replace the patch with a standalone Cargo.lock file, which is easier to read and generate. See `pkgs/development/libraries/rure/default.nix` for examples.